presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1047,6 +1052,25 @@ private OuterJoinPushDownResult processLimitedOuterJoin(RowExpression inheritedP
                     postJoinConjuncts.add(conjunct);
                 }
             }
+
+            if (SystemSessionProperties.shouldGenerateDomainFilters(session)) {
+                // Extract domains for each of the variables from the inherited predicate
+                // See related comment on #processInnerJoin
+                rowExpressionDomainTranslator.fromPredicate(session.toConnectorSession(), inheritedPredicate)
+                        .getTupleDomain()
+                        .getDomains()
+                        .ifPresent(map -> map.forEach((variable, domain) -> {
+                            // For outer-side, inferred domains can be pushed down as-is
+                            if (outerVariables.contains(variable)) {
+                                outerPushdownConjuncts.add(rowExpressionDomainTranslator.toPredicate(domain, variable));
```

Comment:
maybe something like the following to avoid exposing private functions in RowExpresstionDomainTranslator
outerPushdownConjuncts.add(rowExpressionDomainTranslator.toPredicate(TupleDomain.withColumnDomains(ImmutableMap.of(variable, domain))));

Reply:
I think we can make this single column/single variable `toPredicate` translation public. I can add a Javadoc to make this explicit, and add an interface definition in `DomainTranslator`

Reply:
Is there any benefit to doing this? Are there multiple use cases for such a method?

Reply:
Not at the moment no. Its beneficial by 1/ Avoiding a un-necssary allocations 2/Provides a more readble overload of `toPredicate` for single Domain translation